flow/scripts/variables.yaml

```diff
@@ -0,0 +1,577 @@
+    Strategies for Yosys ABC synthesis: Area/Speed. Default ABC_SPEED.
+  stages:
+    - synth
+PWR_NETS_VOLTAGES:
+  description: >
+    Used for IR Drop calculation.
+  stages:
+    - final
+GND_NETS_VOLTAGES:
+  description: >
+    Used for IR Drop calculation.
+  stages:
+    - final
+BLOCKS:
+  description: >
+    Blocks used as hard macros in a hierarchical flow. Do note that you have to
+    specify block-specific inputs file in the directory mentioned by Makefile.
+CDL_FILES:
+  description: >
+    Insert additional Circuit Description Language (`.cdl`) netlist files.
+DFF_LIB_FILES:
+  description: >
+    Technology mapping liberty files for flip-flops.
+  stages:
+    - final
+DONT_USE_LIBS:
+  description: >
+    Set liberty files as `dont_use`.
+PRESERVE_CELLS:
+  description: >
+    Mark modules to keep from getting removed in flattening.
+SYNTH_ARGS:
+  description: >
+    Optional synthesis variables for yosys.
+  stages:
+    - final
+VERILOG_TOP_PARAMS:
+  description: >
+    Apply toplevel params (if exist).
+  stages:
+    - final
+CORE_ASPECT_RATIO:
+  description: >
+    The core aspect ratio (height / width). This value is ignored if `CORE_UTILIZATION`
+    is undefined.
+  stages:
+    - floorplan
+CORE_MARGIN:
+  description: >
+    The margin between the core area and die area, in multiples of SITE heights.
+    The margin is applied to each side. This variable is ignored if `CORE_UTILIZATION`
+    is undefined.
+  stages:
+    - floorplan
+DIE_AREA:
+  description: >
+    The die area specified as a list of lower-left and upper-right corners in microns
+    (X1 Y1 X2 Y2).
+  stages:
+    - floorplan
+RESYNTH_AREA_RECOVER:
+  description: >
+    Enable re-synthesis for area reclaim.
+  stages:
+    - synth
+RESYNTH_TIMING_RECOVER:
+  description: >
+    Enable re-synthesis for timing optimization.
+  stages:
+    - synth
+MACRO_HALO_X:
+  description: >
+    Set macro halo for x-direction. Only available for ASAP7 PDK.
+  stages:
+    - floorplan
+MACRO_HALO_Y:
+  description: >
+    Set macro halo for y-direction. Only available for ASAP7 PDK.
+  stages:
+    - floorplan
+MACRO_WRAPPERS:
+  description: >
+    The wrapper file that replaces existing macros with their wrapped version.
+  stages:
+    - floorplan
+CTS_BUF_DISTANCE:
+  description: >
+    Distance (in microns) between buffers.
+  stages:
+    - cts
+CTS_CLUSTER_DIAMETER:
+  description: >
+    Maximum diameter (in microns) of sink cluster. Default 20.
+  stages:
+    - cts
+CTS_CLUSTER_SIZE:
+  description: >
+    Maximum number of sinks per cluster. Default 50.
+  stages:
+    - cts
+CTS_SNAPSHOT:
+  description: >
+    Creates ODB/SDC files prior to clock net and setup/hold repair.
+  stages:
+    - cts
+POST_CTS_TCL:
+  description: >
+    Specifies a Tcl script with commands to run after CTS is completed.
+  stages:
+    - cts
+FASTROUTE_TCL:
+  description: >
+    Specifies a Tcl script with commands to run before FastRoute.
+  stages:
+    - grt
```

Comment:
I thought you would leave the stages as all by default.  This is needed in global_place.tcl too (and gui.tcl).

Reply:
yes, but I was looking at the documentation sections and pondering how to best represent things...

Reply:
KISS: refine stages later and leave variables in "all" section when stages section is not carefully curated.